views/orders.py

```diff
@@ -104,13 +112,18 @@ def put(self, order_id):
             if i not in res['order']:
                 return None, BAD_REQUEST
 
+        # check whether availabilities allow order update
+        if any(item.availability < res_items[item.name]['quantity']
+                for item in items):
+            return None, BAD_REQUEST
+
         # Clear the order of all items before adding the new items
         # that came with the PUT request
         order.empty_order()
 
-        for item in res['order']['items']:
+        for name, item in res_items.items():
             order.add_item(
-                Item.get(Item.name == item['name']), item['quantity'])
+                Item.get(Item.name == name), item['quantity'])
```

Comment:
evitiamo questo loop e usiamo il risultato della query di prima (che ovviamente andrà salvato in una lista)